pkg/validate/validate.go

```diff
@@ -107,25 +107,63 @@ func ImagePullSecret(secret string) error {
 // the ClusterName function to confirm compliance with the rest. The resulting name will
 // therefore match RFC1035 with the exception of possible periods ".", which will be
 // translated into dashes "-" in the InfraID before being used to create cloud resources.
-func ClusterName1035(v string) error {
-	re := regexp.MustCompile("^[a-z]")
-	if !re.MatchString(v) {
-		return errors.New("cluster name must begin with a lower-case letter")
+// The optional param m is the max length for the string which the platform support.
+func ClusterName1035(v string, m ...int) error {
+
+	// checks the cluster name starts with a lower-case letter
+	err := ClusterStartName(v)
+	if err != nil {
+		return err
 	}
-	return ClusterName(v)
+
+	// checks if max length for the string was set
+	if m != nil {
+		err := ClusterName(v, m...)
+		if err != nil {
+			return err
+		}
+	} else {
+		err := ClusterName(v)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
```

Comment:
We should get rid of the variadic function, but for future reference, you do not need to have this conditional logic.
```suggestion
	return ClusterName(v, m...)
```